compiler/rustc_ty_utils/src/layout.rs

```diff
@@ -105,11 +105,13 @@ fn map_error<'tcx>(
             // See `tests/ui/layout/trivial-bounds-sized.rs` for an example.
             assert!(field.layout.is_unsized(), "invalid layout error {err:#?}");
             if !field.ty.is_sized(cx.tcx(), cx.typing_env) {
-                cx.tcx().dcx().delayed_bug(format!(
+                let guar = cx.tcx().dcx().delayed_bug(format!(
                     "encountered unexpected unsized field in layout of {ty:?}: {field:#?}"
                 ));
+                LayoutError::ReferencesError(guar)
+            } else {
+                LayoutError::Unknown(ty)
             }
-            LayoutError::Unknown(ty)
         }
         LayoutCalculatorError::EmptyUnion => {
             // This is always a compile error.
```

Comment:
Can you do the same for the two delayed bugs below?

They are for these, which i suspect will still ICE:

```rust
union Foo {}

enum Bar {
    Boo = {
        let _: Option<Foo> = None;
        0
    },
}
```

```rust
#[repr(u8, packed)]
enum Foo {
    A
}

enum Bar {
    Boo = {
        let _: Option<Foo> = None;
        0
    },
}
```